src/App.tsx

```diff
@@ -1,156 +1,84 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { ErrorMesage } from './components/ErrorMessage/ErrorMesage';
+import { Filter } from './types/Filter';
+import { Todo } from './types/Todo';
+import { fetchTodos } from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 11028;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(Filter.ALL);
+  const [errorNotification, setErrorNotification] = useState(false);
+  const [error, setError] = useState('');
+  const [search, setSearch] = useState('');
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const arrTodos = await fetchTodos(USER_ID.toString());
+
+        setTodos(arrTodos);
+        setErrorNotification(false);
+      } catch {
+        setErrorNotification(true);
+        setError('Unable to fetch todos');
+      }
+    };
+
+    if (USER_ID) {
+      fetchData();
+    }
+  }, [USER_ID, todos]);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filter) {
+        case Filter.ACTIVE:
+          return !todo.completed;
+        case Filter.COMPLETED:
+          return todo.completed;
+        default:
+          break;
+      }
+
+      return true;
```

Comment:
move filter inside switch case no vice versa too avoid extra iteration while filter is ALL